temporal-sdk/src/main/java/io/temporal/common/metadata/POJOWorkflowImplMetadata.java

```diff
@@ -238,4 +243,35 @@ public List<POJOWorkflowMethodMetadata> getUpdateValidatorMethods() {
   public @Nullable Constructor<?> getWorkflowInit() {
     return workflowInit;
   }
+
+  /**
+   * @return The {@link VersioningBehavior} for the workflow method on the implementation class. If
+   *     the method is annotated with {@link WorkflowVersioningBehavior}.
+   * @throws RuntimeException if the method is not found on the implementation class or is not a
+   *     workflow method.
+   */
+  @Experimental
+  @Nullable
+  public static VersioningBehavior getVersioningBehaviorForMethod(
```

Comment:
We may want to make this a bit more generic if we plan to add other annotations to workflow impl. methods. Since this is experimental I am fine leaving it as is.